### ISSUE TITLE
New version doesn’t see updates anymore

### ISSUE THREAD

**Issue by flying-sheep** [aurapm/aura#408]
so the last version detected the updates but then couldn’t proceed:

```console
$ sudo aura -Au
aura >>= Fetching package information...
aura >>= Comparing package versions...
aura >>= AUR Packages to upgrade:
apache-spark : 1.6.0-5 => 1.6.1-1
atom-editor : 1.5.4-1 => 1.6.1-1
aura-bin : 1.3.4-1 => 1.3.5-1
cower : 14-2 => 16-1
dropbox : 3.14.7-1 => 3.16.1-1
openblas-lapack : 0.2.15-4 => 0.2.17-1
powerpill : 2016.3-1 => 2016.3.17-1
pycharm-community : 5.0.4-1 => 2016.1-1
qt5-quick1 : 5.5.1-10 => 5.6.0-2
rstudio-desktop-bin : 0.99.891-1 => 0.99.893-1
seafile-client-cli : 5.0.5-1 => 5.0.5-2
seafile-shared : 5.0.5-1 => 5.0.5-2
ucsc-kent-genome-tools : 326-1 => 327-1
viber : 4.2.2.6-14 => 4.2.2.6-17
zotero : 4.0.29.2-1 => 4.0.29.5-1

aura >>= The following are not AUR packages:
apache-spark
atom-editor
aura-bin
cower
dropbox
openblas-lapack
powerpill
pycharm-community
qt5-quick1
rstudio-desktop-bin
seafile-client-cli
seafile-shared
ucsc-kent-genome-tools
viber
zotero

aura >>= No valid packages specified.
```

and the current one can’t detect them:

```console
$ sudo aura -Au
aura >>= Fetching package information...
aura >>= Comparing package versions...
aura >>= No AUR package upgrades necessary.
aura >>= No valid packages specified.
$ aura -Qi pycharm-community | grep Version
Version : 5.0.4-1
$ aura -Ai pycharm-community | grep Version
Version : 2016.1-1
```

**Comment by ThomasWeiser** [comment]
I can confirm this, exactly same issue.

**Comment by somebears** [comment]
Can confirm aswell

**Comment by jplatte** [comment]
This issue was already fixed a week ago. You'll have to update aura manually.

**Comment by ThomasWeiser** [comment]
> This issue was already fixed a week ago. You'll have to update aura manually.

IMHO this may be a different issue.
I updated `aura-bin` manually to `aura-bin-1.3.5-1`.
This version doesn't detect pending updates.

If I am supposed to rebuild some cache or the like please tell me how to do so. I am unsure how aura determines those packages that were previously built and installed by it.

**Comment by jplatte** [comment]
Well, what @flying-sheep described is the bug that was fixed a week ago, where checking for updates and searchined worked, but downloading PKGBUILDs or something along that line fails.

If `aura -Au` doesn't show any updates but you know there are some packages in the AUR in newer versions than you have installed locally, then you might want to create a new issue. It works for me, though (I have the same package version). There is no cache to be cleared.

**Comment by ThomasWeiser** [comment]
I'm not getting it. @flying-sheep introduced this issue with a description of the old problem (issue #404) and then continues to describe the new problem.

@flying-sheep Do I misunderstand your issue? If not, maybe you should focus in your opening post on the new problem.

**Comment by flying-sheep** [comment]
actually i opened this issue the day it occurred so it can’t still apply today.

i identified it 2016-03-30 and you said it would’ve been fixed “a week ago” i.e. 3-31 - 7 = ~ 3-24

**Comment by jplatte** [comment]
@ThomasWeiser: Oh right. Guess I didn't read the original post thoroughly enough, sorry.

One thing I see with the package you gave as an example though is that the versioning scheme changed completely. Like I said I upgraded a package (I think it was `gtk-theme-arc-git`) today just fine and that worked, so either it's specific packages only, or specific circumstances.

**Comment by ThomasWeiser** [comment]
Don't think that it's specific to certain packages or their versioning scheme. Here is a list of pending updates on my machine, as determined by former version 1.3.4 (patched by replacing `aur4.archlinux.org` with `aur.archlinux.org` in the binary):

```
> aura-1.3.4-patched -Au
atom-editor-bin : 1.6.0-1 => 1.6.1-1
firefox-aurora : 47.0a2.20160325004031-1 => 47.0a2.20160331004021-1
freefilesync : 8.0-0 => 8.0-1
hugo : 0.13-1 => 0.15-1
ncurses5-compat-libs : 6.0-2 => 6.0-3
webstorm : 2016.1-4 => 2016.1.1-1
```

The current version 1.3.5 doesn't list any of these updates:

```
> aura -Au
aura >>= Fetching package information...
aura >>= Comparing package versions...
aura >>= No AUR package upgrades necessary.
aura >>= No valid packages specified.
```
```
> aura --version                      
 __ _ _  _ _ _ __ _    Pacman v5.0.1 - libalpm v10.0.1
/ _` | || | '_/ _` |   Copyright (C) 2006-2016 Pacman Development Team
\__,_|\_,_|_| \__,_|   Copyright (C) 2002-2006 Judd Vinet
AURA Version 1.3.5     
```

Thank you for attending to the matter!

**Comment by jplatte** [comment]
> Thank you for attending to the matter!

I didn't really do anything besides getting confused about what this bug report is actually about and doing some basic troubleshooting. And I'm afraid that might be all I am able to do without being able to reproduce this on my end. Would be good to know whether @fosskers can reproduce this...

But to go forward with the basic troubleshooting, I did find the option `--auradebug`. For me `sudo aura --auradebug -Au` outputs:

```
User              => root
True User         => jplatte
Build User        => jplatte
Using Sudo?       => Yes!
Pacman Flags      => 
Other Flags       => Debug AURInstall Upgrade
Other Input       => 
Language          => English
Pacman Command    => pacman
Editor            => vi
$CARCH            => x86_64
Ignored Pkgs      => 
Build Path        => /var/cache/pacman/pkg/
Pkg Cache Path    => /var/cache/pacman/pkg/
Log File Path     => /var/log/pacman.log
Quiet?            => No.
Silent Building?  => Yes!
Must Confirm?     => Yes!
Needed only?      => No.
PKGBUILD editing? => No.
Diff PKGBUILDs?   => No.
Rebuild Devel?    => No.
Use Customizepkg? => No.
Forego PowerPill? => No.
Keep source?      => No.
```

before the usual output. Are there any differences if one of you experiencing this bug run that?

**Comment by ThomasWeiser** [comment]
I have the same output from `--auradebug`

**Comment by huyderman** [comment]
I'm having the same problem on my laptop, but not on my desktop. I can't remember what I did different when upgrading to 1.3.5 the two different computers.

**Comment by AlexTalker** [comment]
This is wors, folks:
```
sudo aura -Au
aura >>= Fetching package information...
aura >>= Comparing package versions...
aura >>= AUR Packages to upgrade:
2gis : 3.15.10.0-1 => 3.16.3.0-1
aarchup : 1.7.3-2 => 1.8.1-1
alsaequal : 0.6-12 => 0.6-14
cargo-bin : 20160119-1 => 20160216-1
chromium-pepper-flash-standalone : 20.0.0.286-1 => 21.0.0.197-1
copyq-git : 20141026-1 => 20151130-1
cppreference-qt : 20150808-1 => 20151129-1
create_ap : r160.0f5a503-1 => r191.69541cb-1
dbvis : 9.2.13-1 => 9.2.15-1
decklink-sdk : 1:10.1.4-1 => 1:10.5-1
eclipse-checkstyle : 6.5.0-1 => 6.16.0-1
eiskaltdcpp-core-git : 2.3.0.4848.d06b149-1 => 2.3.0.4927.126f287-1
eiskaltdcpp-daemon-git : 2.3.0.4848.d06b149-1 => 2.3.0.4927.126f287-1
eiskaltdcpp-data-git : 2.3.0.4848.d06b149-1 => 2.3.0.4927.126f287-1
epsxe : 1.9.25-8 => 1.9.25-10
fortune-mod-vimtips : 20131101-2 => 201510719-3
freshplayerplugin : 0.3.4-1 => 0.3.4-2
htmlcxx : 0.85-4 => 0.86-1
inox-bin : 48.0.2564.82-1 => 49.0.2623.110-1
intellij-idea-ce-eap : 3:16.144.3891.8-1 => 2016.1.145.596.7-1
jdk : 8u51-2 => 8u77-1
jdk7 : 7.80-2 => 7u79-4
jitsi : 2.8.5426-2 => 2.8.5426-3
jmk-x11-fonts : 3.0-1 => 3.0-2
jre : 8u51-2 => 8u77-1
jre7 : 7u79-1 => 7u79-3
lib32-libcurl-compat : 7.23.1-2 => 7.47.1-1
lib32-libpng12 : 1.2.56-1 => 1.2.56-2
lib32-libwebp : 0.4.4-1 => 0.5.0-1
lib32-sdl2_image : 2.0.0-1 => 2.0.1-1
libbs2b : 3.1.0-2 => 3.1.0-4
libdbusmenu-glib : 12.10.2-4 => 16.04.0-1
libdbusmenu-gtk3 : 12.10.2-4 => 16.04.0-1
libnatspec : 0.2.6-1 => 0.3.0-2
libpng12 : 1.2.56-1 => 1.2.56-2
libqtxdg-git : 1.1.0.29.g641754a-1 => 1.2.0.13.gab0f65c-1
libutvideo-git : v15.0.2.92.ffc9061-1 => v15.1.0.64.g36eb60c-1
megatools : 1.9.95-3 => 1.9.97-1
ncurses5-compat-libs : 6.0-2 => 6.0-3
otf-font-awesome : 4.3.0-2 => 4.5.0-1
packettracer : 6.2-2 => 6.3-1
pamixer-git : 20121122-3 => 20150209-1
popcorntime-git : r5144.6444037-1 => r5880.70671ae-1
profanity-git : 2827.8e46b9e-1 => 4170.56751e8-1
psi-plus-git : 0.16.457-1 => 0.16.475-1
psi-plus-l10n-git : 0.16.441.2-1 => 0.16.457-1
psi-plus-plugins-git : 0.15.443-1 => 0.15.448-1
python-aiodns : 0.3.1-1 => 1.0.0-1
python-pycares : 0.6.3-1 => 1.0.0-1
python-slixmpp-git : r1796.62eefdb-1 => r2032.b549db9-1
python-tomate : 0.4.1-1 => 0.5.0-1
python32 : 3.2.6-1 => 3.2.6-2
qbittorrent-git : 3.2.5.r0.gc0ccf28-1 => 3.3.1.r141.049e622-1
qownnotes : 0.67-1 => 1.3.3-1
qtermwidget-git : 0.6.0.46.g0e50464-1 => 0.6.0.48.g10e1796-1
qtkeychain-qt4 : 0.5-2 => 0.6-1
qxkb : 0.4.6-2 => 0.4.6-3
recstudio : 4.0_20130724-1 => 4.1_20150924-1
retroshare : 0.5.5c-2 => 0.6.0-1
rtmidi : 2.1.0-3 => 2.1.1-1
ruby2.1 : 2.1.6-1 => 2.1.9-1
signify : 8-1 => 17-1
teamviewer : 11.0.52520-1.5 => 11.0.53191-2.3
thermald : 1.4.3-1 => 1.5.3-1
tomate-gtk : 0.4.0-1 => 0.5.0-1
unzip-natspec : 6.0-5 => 6.0-6
veracrypt : 1.16-6 => 1.17-2
vim-toml-git : 20141015-1 => 20150403-1
visual-paradigm-community : 13.0.20160103-1 => 13.0.20160304-1
weechat-git : 1:1.3.52.g0f40ca1-1 => 1.3.5.gd0b11ec-1

aura >>= The following are not AUR packages:
eiskaltdcpp-core-git
eiskaltdcpp-daemon-git
eiskaltdcpp-data-git
libdbusmenu-glib
libdbusmenu-gtk3

aura >>= Determining dependencies...
aura >>= Dependency checking failed for these reasons:
The dependency `qt5-multimediaqt5-x11extraslibupnplibgnome-keyringlibxsslibmicrohttpdsqlcipherffmpegopencv` could not be found. You may need to search for a package to satisfy it.
```

**Comment by fosskers** [comment]
Hey guys, with `1.3.5` I just successfully performed an `-Au`.

**Comment by ThomasWeiser** [comment]
@fosskers: Thanks for testing. On my machine the problem still exists: While there are a set of necessary package updates, `aura -Au` doesn't find any (version aura-bin 1.3.5-1). Installing or updating packages with `aura -A <package>`work fine.

So it seems to work for some people, but not for others. Only idea I have at the moment is, that I used the patched version 1.3.4 in the meanwhile, which others may have not. Maybe this makes a difference.

I am curious: how does aura detect necessary updates? Does it compare the whole list of installed packages with the AUR? Or is there a list of packages formerly installed by `aura -A`? Anything I can check on my side?

**Comment by jplatte** [comment]
@ThomasWeiser I think the best idea is to bisect this issue so we can find out where this bug was introduced. Seems like 98c61b3d7e1cbb8066349c18d36cccdfcce7206c is 1.3.4 and 654cfd7952ad04bd347364e2e1ae41879bdbd65c is 1.3.5. Would have been easier if this repo had release tags by the way, @fosskers.

Anyway, if you're unfamiliar with git bisect, here's basically what you have to do:

```
git checkout 98c61b3d7e1cbb8066349c18d36cccdfcce7206c
stack build
stack exec sudo aura -- -Au

git checkout 654cfd7952ad04bd347364e2e1ae41879bdbd65c
stack build
stack exec sudo aura -- -Au
```

and then, assuming you still get the same results (1.3.4 detects updates, 1.3.5 doesn't)

```
git bisect start
git bisect good 98c61b3d7e1cbb8066349c18d36cccdfcce7206c
git bisect bad # current commit
```

which will make git select and checkout some commit between 1.3.4 so you can check whether the error is present there, and mark that with `git bisect good` (error not present) or `git bisect bad` (error present). With the amount of commits between 1.3.4 and 1.3.5 (17), I think you should have to do this maximally 5 times before git tells you which commit introduced the bug.

**Comment by ThomasWeiser** [comment]
I would gladly do a git bisect. Unfortunately my Haskell installation is broken at the moment. I am unable to repair it due to external constraints. If the problem still exists until then, I may be able to try it in two weeks.

Maybe someone else who reported the issue here can jump in.

**Comment by sebeaumont** [comment]
I'm seeing exactly this issue with aura built from the github head today -- just doesn't find any packages at all... I m using ghc 7.10.3 and cabal 1.22 to build -- unfortunately I can't spend time on this -- need to install some stuff and get going.

**Comment by fosskers** [comment]
Hey guys, thanks for your patience.

@ThomasWeiser the command `pacman -Qm` will tell you what foreign packages
exist on your system. Aura uses this information, does a batch query on
package names, and compares local and upstream version numbers for
updates.

@jplatte re: tags. Yes, you're right. I haven't been as vigilant with that
lately.

@sebeaumont Unfortunately I'm able to `-Au` with Aura built from the
github head.

Everyone, the problem with 1.3.4 existed in the `aur` package. 1.3.5
is only a small change to handle an update to `aur`. If there is a problem,
I think it's there and not within Aura itself.

**Comment by jacksenechal** [comment]
From the aura-bin AUR page:

> **anks commented on 2016-03-21 19:15**
> In case you were getting the following error:
> 
> aura >>= The following are not AUR packages ...
> 
> Here's some hackish fix:
> 
> `sudo sed -i 's/https:\/\/aur4.archlinux.org\//https:\/\/aur.archlinux.org\/\x00/' /usr/bin/aura`

I was having this issue and the above sed command was a good quick fix. Then running `sudo aura -A aura-bin` works like normal.

**Comment by shioyama** [comment]
@jacksenechal +1 thanks for the tip!

**Comment by joshproehl** [comment]
I have two computers with aura-bin installed (1.3.5). On one of them Aura _seems_ to be working correctly, but on the other all I get is:
> aura >>= Fetching package information...
> aura >>= Comparing package versions...
> aura >>= No AUR package upgrades necessary.
> aura >>= No valid packages specified.

I've uninstalled aura-bin and re-installed it manually, to no effect. The above sed replacement doesn't seem to have any effect either.

Any suggestions or things I can do to help debugging of this issue? (Other than git bisect, which I _may_ be able to attempt if necessary.)

**Comment by fosskers** [comment]
@joshproehl Do you have a Haskell dev environment?

**Comment by joshproehl** [comment]
@fosskers I was able to compile the current HEAD using stack, and it exhibits exactly the same behavior as the installed package.
(An example package which should be caught is Spotify, currently version 1.0.27.73-1 in AUR. `pacman -Qm spotify` shows spotify 1.0.25.127-1)

Checking out 98c61b3d7e1cbb8066349c18d36cccdfcce7206c, with a minor change to stack.yaml to pull aur 4.0.2, is partially successful (at least it finds that there are things TO upgrade, possibly the aur4 problem?):
> $ sudo "$(stack path --local-install-root)/bin/aura" -A
> aura >>= Fetching package information...
> aura >>= Comparing package versions...
> aura >>= AUR Packages to upgrade:
> android-sdk : 24.4.1-1 => 24.4.1-4
> dropbox : 3.16.1-1 => 3.18.1-1
> eclipse-gef : 3.10.0-1 => 3.10.1-1
> firefox-aurora : 47.0a2.20160326004034-1 => 47.0a2.20160417004052-1
> fontconfig-ubuntu : 2.11.1-3 => 2.11.94-1
> freetype2-ubuntu : 2.5.2-8 => 2.6.1-1
> haskell-aur : 4.0.2-1 => 5.0.0-1
> isl : 0.15-1 => 0.16-1
> jitsi : 2.8.5426-1 => 2.8.5426-3
> laptop-mode-tools : 1.69.1-1 => 1.69.2-1
> libcurl-compat : 7.47.0-1 => 7.48.0-1
> qtkeychain-qt4 : 0.5-2 => 0.6.1-1
> spotify : 1.0.25.127-1 => 1.0.27.73-1
> yed : 3.14.4-1 => 3.15.0.2-1
> 
> aura >>= The following are not AUR packages:
> android-sdk
> dropbox
> eclipse-gef
> firefox-aurora
> fontconfig-ubuntu
> freetype2-ubuntu
> haskell-aur
> isl
> jitsi
> laptop-mode-tools
> libcurl-compat
> qtkeychain-qt4
> spotify
> yed
> 
> aura >>= No valid packages specified.

Bisecting between 98c61b3d7e1cbb8066349c18d36cccdfcce7206c and current HEAD (this timestamp) reveals:

> 82161de5886734485b697c3b54e43576fb54c733 is the first bad commit
> commit 82161de5886734485b697c3b54e43576fb54c733
> Author: Colin Woodbury <colingw@gmail.com>
> Date:   Thu Mar 24 13:09:36 2016 -0700
> 
>     Use `aur-5.0.0` - fixes #404
> 
> :100644 100644 35874b8e1fb84ef3785f216f6dd17c0ba1824921 769ba6bda966cac154911dee6f0d418227f28f12 M	PKGBUILD
> :100644 100644 c6df791ce89e7f719b7c4f64b3a3825934ad53ec 6cc315afe0da0ba398439463f5c6bf2dcb93128b M	aura.cabal
> :040000 040000 b352a777e3df547c69136450d86f62bdc0102352 3ae513483efe11add2c1688f00451429e2b11f38 M	src
> :100644 100644 a2e657c6ba22b90c478766245fd69dbe7ce41955 3638199cb465706076514b246e835d4d9d62da3d M	stack.yaml

(Also, to confirm, 82161de5886734485b697c3b54e43576fb54c733 was the first commit that failed to identify that there were packages in need of upgrade)

Sorry for the delayed reply. Please let me know if I can be of further assistance.

**Comment by illis** [comment]
Can confirm I get the same issue as @joshproehl. 

https://github.com/aurapm/aura/commit/514f54ea98462bc750a5434c3c2ec6b69a4e317b Gives me similar output (different packages) to what @joshproehl posted above.

https://github.com/aurapm/aura/commit/82161de5886734485b697c3b54e43576fb54c733 Doesn't detect any package changes.

Still learning Haskell, so my coding skills aren't quite up to scratch; but willing to help out testing any patches/branches.

**Comment by fosskers** [comment]
Quite update as I'm fixing this: `aur` seems to have broken again, strangely. Its `info` and `search` calls to the AUR seem to be timing out.

**Comment by flying-sheep** [comment]
so aura 1.3.6 still doesn’t work. i upgraded powerpill and all.

**Comment by fosskers** [comment]
Can you try the following each several times, and tell me the result?

```
curl -4 "https://aur.archlinux.org/rpc?v=5&type=search&arg=foobar"
curl -6 "https://aur.archlinux.org/rpc?v=5&type=search&arg=foobar"
```
They constrain the request to use ipv4 and 6 respectively.

**Comment by joshproehl** [comment]
@fosskers here are the results from one run of each. I did each several times as you requested, they appear to be the same on each run. (I am running a dual-stack network.)

> curl -4 "https://aur.archlinux.org/rpc?v=5&type=search&arg=foobar"
> {"version":5,"type":"search","resultcount":3,"results":[{"ID":292513,"Name":"foobar2000","PackageBaseID":37068,"PackageBase":"foobar2000","Version":"1.3.10-1","Description":"An advanced freeware audio player (uses Wine).","URL":"http:\/\/www.foobar2000.org\/","NumVotes":39,"Popularity":0.10115,"OutOfDate":null,"Maintainer":"supermario","FirstSubmitted":1273255356,"LastModified":1460211786,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000.tar.gz"},{"ID":293480,"Name":"foobar2000-playcount-bin","PackageBaseID":110061,"PackageBase":"foobar2000-playcount-bin","Version":"3.0.2-1","Description":"Collects playback statistics for your music","URL":"https:\/\/www.foobar2000.org\/components\/view\/foo_playcount","NumVotes":0,"Popularity":0,"OutOfDate":null,"Maintainer":"doskoi","FirstSubmitted":1460512003,"LastModified":1460513634,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000-playcount-bin.tar.gz"},{"ID":293492,"Name":"foobar2000-texttools-bin","PackageBaseID":110063,"PackageBase":"foobar2000-texttools-bin","Version":"1.0.5-1","Description":"Adds customizable context menu commands for copying information about the selected tracks to clipboard","URL":"https:\/\/www.foobar2000.org\/components\/view\/foo_texttools","NumVotes":0,"Popularity":0,"OutOfDate":null,"Maintainer":"doskoi","FirstSubmitted":1460515326,"LastModified":1460515667,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000-texttools-bin.tar.gz"}]}% 

> curl -6 "https://aur.archlinux.org/rpc?v=5&type=search&arg=foobar"
> {"version":5,"type":"search","resultcount":3,"results":[{"ID":292513,"Name":"foobar2000","PackageBaseID":37068,"PackageBase":"foobar2000","Version":"1.3.10-1","Description":"An advanced freeware audio player (uses Wine).","URL":"http:\/\/www.foobar2000.org\/","NumVotes":39,"Popularity":0.10115,"OutOfDate":null,"Maintainer":"supermario","FirstSubmitted":1273255356,"LastModified":1460211786,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000.tar.gz"},{"ID":293480,"Name":"foobar2000-playcount-bin","PackageBaseID":110061,"PackageBase":"foobar2000-playcount-bin","Version":"3.0.2-1","Description":"Collects playback statistics for your music","URL":"https:\/\/www.foobar2000.org\/components\/view\/foo_playcount","NumVotes":0,"Popularity":0,"OutOfDate":null,"Maintainer":"doskoi","FirstSubmitted":1460512003,"LastModified":1460513634,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000-playcount-bin.tar.gz"},{"ID":293492,"Name":"foobar2000-texttools-bin","PackageBaseID":110063,"PackageBase":"foobar2000-texttools-bin","Version":"1.0.5-1","Description":"Adds customizable context menu commands for copying information about the selected tracks to clipboard","URL":"https:\/\/www.foobar2000.org\/components\/view\/foo_texttools","NumVotes":0,"Popularity":0,"OutOfDate":null,"Maintainer":"doskoi","FirstSubmitted":1460515326,"LastModified":1460515667,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000-texttools-bin.tar.gz"}]}%

**Comment by fosskers** [comment]
Thanks. For me it cuts in and out. Can you wait a bit and try again?

Also try through a browser, with the `Network` tab open in the dev tools. Can you see what IP is being called for the AUR?

**Comment by joshproehl** [comment]
Chromium getting the aur query has:
Remote Address:[2a01:4f8:160:3033::2]:443

Which is the only ipv6 address available for aur, according to `host`

I'll try fetching it via curl periodically, but it's been successful every time. The only thing I'm noticing is that the fetch time (in Chromium)is usually 200ms, but occasionally jumps to 700ms.

**Comment by fosskers** [comment]
What country do you live in?

**Comment by joshproehl** [comment]
Portland Oregon, USA. IPv6 is coming through a Hurricane Electric tunnel to Seattle Washington USA.

**Comment by kudlatyamroth** [comment]
i have same issue,
curl ipv4:
`{"version":5,"type":"search","resultcount":3,"results":[{"ID":292513,"Name":"foobar2000","PackageBaseID":37068,"PackageBase":"foobar2000","Version":"1.3.10-1","Description":"An advanced freeware audio player (uses Wine).","URL":"http:\/\/www.foobar2000.org\/","NumVotes":39,"Popularity":0.10098,"OutOfDate":null,"Maintainer":"supermario","FirstSubmitted":1273255356,"LastModified":1460211786,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000.tar.gz"},{"ID":293480,"Name":"foobar2000-playcount-bin","PackageBaseID":110061,"PackageBase":"foobar2000-playcount-bin","Version":"3.0.2-1","Description":"Collects playback statistics for your music","URL":"https:\/\/www.foobar2000.org\/components\/view\/foo_playcount","NumVotes":0,"Popularity":0,"OutOfDate":null,"Maintainer":"doskoi","FirstSubmitted":1460512003,"LastModified":1460513634,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000-playcount-bin.tar.gz"},{"ID":293492,"Name":"foobar2000-texttools-bin","PackageBaseID":110063,"PackageBase":"foobar2000-texttools-bin","Version":"1.0.5-1","Description":"Adds customizable context menu commands for copying information about the selected tracks to clipboard","URL":"https:\/\/www.foobar2000.org\/components\/view\/foo_texttools","NumVotes":0,"Popularity":0,"OutOfDate":null,"Maintainer":"doskoi","FirstSubmitted":1460515326,"LastModified":1460515667,"URLPath":"\/cgit\/aur.git\/snapshot\/foobar2000-texttools-bin.tar.gz"}]}`

curl ipv6:
`url: (7) Couldn't connect to server`

**Comment by fosskers** [comment]
https://github.com/haskell-servant/servant/issues/486

**Comment by flying-sheep** [comment]
```json
$ curl -4 "https://aur.archlinux.org/rpc?v=5&type=search&arg=foobar"
url: (7) Couldn't connect to server
$ curl -4 "https://aur.archlinux.org/rpc?v=5&type=search&arg=foobar" | jshon                                                                                        ⏎
{
 "version": 5,
 "type": "search",
 "resultcount": 3,
 "results": [
  {
   "ID": 292513,
   "Name": "foobar2000",
   "PackageBaseID": 37068,
   "PackageBase": "foobar2000",
   "Version": "1.3.10-1",
   "Description": "An advanced freeware audio player (uses Wine).",
   "URL": "http:\/\/www.foobar2000.org\/",
   "NumVotes": 39,
   "Popularity": 0.10098,
   "OutOfDate": null,
   "Maintainer": "supermario",
   "FirstSubmitted": 1273255356,
   "LastModified": 1460211786,
   "URLPath": "\/cgit\/aur.git\/snapshot\/foobar2000.tar.gz"
  },
  {
   "ID": 293480,
   "Name": "foobar2000-playcount-bin",
   "PackageBaseID": 110061,
   "PackageBase": "foobar2000-playcount-bin",
   "Version": "3.0.2-1",
   "Description": "Collects playback statistics for your music",
   "URL": "https:\/\/www.foobar2000.org\/components\/view\/foo_playcount",
   "NumVotes": 0,
   "Popularity": 0,
   "OutOfDate": null,
   "Maintainer": "doskoi",
   "FirstSubmitted": 1460512003,
   "LastModified": 1460513634,
   "URLPath": "\/cgit\/aur.git\/snapshot\/foobar2000-playcount-bin.tar.gz"
  },
  {
   "ID": 293492,
   "Name": "foobar2000-texttools-bin",
   "PackageBaseID": 110063,
   "PackageBase": "foobar2000-texttools-bin",
   "Version": "1.0.5-1",
   "Description": "Adds customizable context menu commands for copying information about the selected tracks to clipboard",
   "URL": "https:\/\/www.foobar2000.org\/components\/view\/foo_texttools",
   "NumVotes": 0,
   "Popularity": 0,
   "OutOfDate": null,
   "Maintainer": "doskoi",
   "FirstSubmitted": 1460515326,
   "LastModified": 1460515667,
   "URLPath": "\/cgit\/aur.git\/snapshot\/foobar2000-texttools-bin.tar.gz"
  }
 ]
}
```

**Comment by fosskers** [comment]
https://github.com/vincenthz/hs-connection/issues/18

**Comment by fosskers** [comment]
We're digging pretty deep here. Please go upvote the issue I just linked to.

**Comment by fosskers** [comment]
The IPV4 for the AUR seems to be 5.9.250.164, except when you reenter that in a browser's search bar, it resolves to bbs.archlinux.org via IPV6, with a 301 "Moved Permanently".

**Comment by fosskers** [comment]
Firefox claims https://aur.archlinux.org/rpc is at 5.9.250.164:443, with no redirect.

**Comment by fosskers** [comment]
https://bbs.archlinux.org/viewtopic.php?pid=1387264#p1387264

**Comment by flying-sheep** [comment]
that did nothing for me. but i also never had problems with timeouts, it almost instantly reports “No AUR package upgrades necessary”

**Comment by joshproehl** [comment]
FYI, aura 1.3.6 experiences this problem in an IPv4-only environment, so the problem doesn't appear to be IPv6 specific.

**Comment by fosskers** [comment]
Thank you for your patience guys, I know how annoying this is.

**Comment by joshproehl** [comment]
Anything we can do to help diagnose this issue further?

**Comment by fosskers** [comment]
Not at the moment, I'm going to see how things are once we're git-based.

**Comment by joshproehl** [comment]
Checking in on this again to see if there's anything we can do to help. Aura is currently unusable for actually managing AUR packages, and I'd prefer not to try and find a new aur helper as I've been quite happy here! :-)

I'm happy to try and help write some code if there's a need, but at this point I'm not sure what exactly is needed.

**Comment by originalsouth** [comment]
In the mean while could do something ugly like: 
`
for p in $(sudo aura -Qm | cut -d ' ' -f 1); do sudo aura -A --noconfirm $p;done
`
as a workaround...

**Comment by fosskers** [comment]
AUR connection issues fixed with `1.3.7`.
